timemachine/fe/single_topology.py

```diff
@@ -1615,6 +1517,326 @@ def _setup_intermediate_chiral_bond_term(
 
         return ChiralBondRestraint(chiral_bond_idxs, np.array(chiral_bond_signs)).bind(jnp.array(chiral_bond_params))
 
+    def _expand_chiral_volumes_into_angles(self, chiral_idxs):
+        induced_angles = set()
+        for c, i, j, k in chiral_idxs:
+            induced_angles.add(canonicalize_bond((i, c, j)))
+            induced_angles.add(canonicalize_bond((i, c, k)))
+            induced_angles.add(canonicalize_bond((j, c, k)))
+        return induced_angles
+
+    def _idxs_belong_to_core(self, idxs):
+        core_atoms = self.get_core_atoms()
+        return all([x in core_atoms for x in idxs])
+
+    def _idxs_belong_to_dummy_a(self, idxs):
+        dummy_atoms = self.get_dummy_atoms_a()
+        return any([x in dummy_atoms for x in idxs])
+
+    def _idxs_belong_to_dummy_b(self, idxs):
+        dummy_atoms = self.get_dummy_atoms_b()
+        return any([x in dummy_atoms for x in idxs])
```

Comment:
nit: seems confusing that `_idxs_belong_to_core` uses `all`, while `_idxs_belong_to_dummy_a` and `_idxs_belong_to_dummy_b` use `any`. Maybe renaming would clarify?

Reply:
explicitly renamed to:

```
    def all_idxs_belong_to_core(self, idxs):
    def any_idxs_belong_to_dummy_a(self, idxs):
    def any_idxs_belong_to_dummy_b(self, idxs):
```